grails-app/services/org/pih/warehouse/picklist/PicklistService.groovy

```diff
@@ -78,7 +78,7 @@ class PicklistService {
             picklist.save()
         }
 
-        productAvailabilityService.refreshProductsAvailability(orderItem?.order?.origin?.id, [orderItem?.product?.id], false)
+        productAvailabilityService.refreshProductsAvailability(orderItem?.order?.origin?.id, null, false)
```

Comment:
Instead of refreshing product availability for all products within a location, you could take products from the picklist before clearing.

Reply:
I've added refreshing while deleting items from picklist